assayist/processor/loose_artifact_analyzer.py

```diff
@@ -38,123 +43,234 @@ def run(self):
+                    # anyway, since that's what contains the actual thing).
+                    # Let's just claim the file and move on.
+                    log.warning(f'Skipping already-claimed symlink in {archive}: '
+                                f'{relative_filepath}')
+                    self.claim_file(path_to_archive, relative_filepath)
+                    continue
+
+                # If we find it locally don't bother asking Koji about it again.
+                if artifact:
+                    self.conditional_connect(original_artifact.embedded_artifacts, artifact)
+                    self.claim_file(path_to_archive, relative_filepath)
+                    continue
+
+                # Add the file to the batch of things to process. If this happens to
+                # trigger a batch execution, handle the resulting Artifacts.
+                for artifact in self.add_to_and_maybe_execute_batch(loose_artifact,
+                                                                    path_to_archive):
+                    self.conditional_connect(original_artifact.embedded_artifacts, artifact)
+
+            # Wrap up any in-progress batch before moving on to the next archive.
+            for artifact in self.execute_batch_and_return_artifacts():
+                self.conditional_connect(original_artifact.embedded_artifacts, artifact)
 
+    def local_lookup(self, loose_artifact):
+        """
+        Lookup the given file locally to see if we already know about it.
+
+        Uses sha256 checksum to make that determination.
+
+        :param str loose_artifact: The full path to the file in question.
+        :raises FileNotFoundError: if the file could not be found to checksum.
+        :return: The Artifact that we discovered with a local lookup, or None.
+        :rtype: Artifact or None
+        """
+        sha256_checksum = self.checksum(loose_artifact)
+        try:
+            checksum_node = content.Checksum.nodes.first(checksum=sha256_checksum)
+        except content.Checksum.DoesNotExist:
+            return None
+
+        # According to the schema a checksum can be associated with multiple Artifacts, but
+        # according to reality that doesn't make much sense. Just return the "first one".
+        artifacts = checksum_node.artifacts.all()
+        if artifacts:
+            log.info(f'Artifact already in database: {loose_artifact}')
+            return artifacts[0]
+        else:
+            return None
+
+    def unpacked_archives(self):
+        """
+        Generate name and path to every unpacked archive.
+
+        :return: All (archive_name, path_to_archive) tuples. The path includes archive_name.
+        :rtype: Iterable
+        """
+        # Dir of all unpacked content
+        unpacked_content_path = os.path.join(self.input_dir, self.UNPACKED_ARCHIVES_DIR)
         for archive_type in os.listdir(unpacked_content_path):  # 'rpm', 'container_layer', 'maven'
```

Comment:
Archives are actually unpacked to a directory named `non-rpm`.

Reply:
You're just commenting about the comment right? Okay, I can update it.

Reply:
Yes :-)